### PR TITLE
<Crusade> Type-only migration!

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to update types and references related to NFT input data in various files.

### Detailed summary
- Updated types to use `NFTInput` instead of `NFTMetadataInput`
- Updated imports and references related to NFT input data in multiple files

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/components/shared/FileInput.tsx [hunk 1]
```diff
@@ -21,13 +21,14 @@ import {
...
 import { FiImage, FiUpload } from "react-icons/fi";
+import type { FileOrBufferOrString } from "thirdweb/storage";
 import { Button, Text } from "tw-components";
 
 interface IFileInputProps extends BoxProps {
   accept?: Accept;
   setValue: (file: File) => void;
   isDisabled?: boolean;
-  value?: string | File;
+  value?: FileOrBufferOrString | undefined;
```
Comment (jnsdls) [inline apps/dashboard/src/components/shared/FileInput.tsx:31]: do we need `Buffer` here? why?

File: apps/dashboard/src/hooks/useImageFileOrUrl.ts [hunk 2]
```diff
@@ -1,15 +1,16 @@
 import { useMemo } from "react";
+import type { FileOrBufferOrString } from "thirdweb/storage";
 import { isBrowser } from "utils/isBrowser";
 
-export function useImageFileOrUrl(image?: string | File): string {
+export function useImageFileOrUrl(image?: FileOrBufferOrString): string {
```
Comment (jnsdls) [inline apps/dashboard/src/hooks/useImageFileOrUrl.ts:5]: why was the prior typing not working?

File: packages/thirdweb/src/exports/storage.ts [hunk 3]
```diff
@@ -10,3 +10,4 @@ export {
   resolveArweaveScheme,
   type ResolveArweaveSchemeOptions,
 } from "../utils/arweave.js";
+export type { FileOrBufferOrString } from "../storage/upload/types.js";
```
Comment (jnsdls) [inline packages/thirdweb/src/exports/storage.ts:13]: are we *sure* we have to export this?

this is meant to handle things between frontends and backends consistently, however the dashboard should only be interacting with things with `Files` and maybe `string`s, so why do we need to export this for usage in dashboard?

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/hooks/useImageFileOrUrl.ts (hunk 2 above)
Comment (kien-ngo) [reply apps/dashboard/src/hooks/useImageFileOrUrl.ts]: I reverted the change back to `main` and it IS working lol
Last night it was throwing some hard-to-read error which led me to believe I have to use the new type that I exported, for consistency.

(Now that this change is reverted I will close this)

File: packages/thirdweb/src/exports/storage.ts (hunk 3 above)
Comment (kien-ngo) [reply packages/thirdweb/src/exports/storage.ts]: WIll revert this change and work on any issue leftover

**Review by MananTank — APPROVED** [review]

**Review by jnsdls — APPROVED** [review]

**Pull request merged**